### PR TITLE
Make PyArrow cpu_count to correspond NUMBA_NUM_THREADS

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
This PR makes PyArrow treads count equal to Numba threads count.
I do not know which threading layer is used by PyArrow. it is possible that there is no intersection with Numba layers.
By default PyArrow uses all available cpus for reading.

### CODE REVIEW

File: sdc/io/csv_ext.py [hunk 1]
```diff
@@ -398,6 +399,18 @@ def to_varname(string):
     return re.sub(r'\W|^(?=\d)','_', string)
 
 
+def pyarrow_cpu_count_equal_numba_num_treads(func):
+    """Decorator. Set pyarrow cpu_count the same as NUMBA_NUM_THREADS."""
+    def wrapper(*args, **kwargs):
+        cpu_count = pyarrow.cpu_count()
+        pyarrow.set_cpu_count(numba.config.NUMBA_NUM_THREADS)
+        result = func(*args, **kwargs)
```
Comment (Hardcode84) [inline sdc/io/csv_ext.py]: nitpick: if wrapped func raise exception you will miss second call to `set_cpu_count` (but I am not sure numba supports finally)

Comment (AlexanderKalistratov) [reply sdc/io/csv_ext.py]: @Hardcode84 as far as I get it this code is not jitted anyway, so `finally` should work

File: sdc/io/csv_ext.py [hunk 2]
```diff
@@ -398,6 +399,28 @@ def to_varname(string):
...
+        self.old_cpu_count = pyarrow.cpu_count()
+        pyarrow.set_cpu_count(self.cpu_count)
+
+    def __exit__(self, type, value, traceback):
+        pyarrow.set_cpu_count(self.old_cpu_count)
+
+
+def pyarrow_cpu_count_equal_numba_num_treads(func):
+    """Decorator. Set pyarrow cpu_count the same as NUMBA_NUM_THREADS."""
+    def wrapper(*args, **kwargs):
```
Comment (densmirn) [inline sdc/io/csv_ext.py]: Consider using of `functools.wraps` here: https://docs.python.org/3/library/functools.html#functools.wraps.

File: sdc/io/csv_ext.py [hunk 3]
```diff
@@ -24,6 +24,7 @@
 # EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 # *****************************************************************************
 
+import os
```
Comment (densmirn) [inline sdc/io/csv_ext.py]: Do you exactly need this import?

File: sdc/tests/tests_perf/test_perf_read_csv.py [hunk 4]
```diff
@@ -53,14 +54,21 @@ def generate_csv(rows=10**5, columns=10):
...
     def _function():
         start = time.time()
         df = pandas.read_csv(file_name)
         return time.time() - start, df
+
+    if pyarrow:
+        def _function():
+            start = time.time()
+            df = sdc.io.csv_ext.pandas_read_csv(file_name)
+            return time.time() - start, df
```
Comment (densmirn) [inline sdc/tests/tests_perf/test_perf_read_csv.py]: In case if `pyarrow is True` `_function()` is defined twice.

Comment (densmirn) [reply sdc/tests/tests_perf/test_perf_read_csv.py]: Maybe do something like that:
```
if pyarrow:
    def _func():
        # ...
    return _func

def _func():
    # ...
return _func
```

File: sdc/tests/tests_perf/test_perf_read_csv.py [hunk 5]
```diff
@@ -43,12 +43,8 @@ def generate_csv(rows=10**5, columns=10):
...
     r = random.Random(0)  # seed=0
     generate(rows,
-        ['A', 'B', 'C'],
-        [
-            lambda: int(r.random() * 10000),
-            lambda: r.uniform(-1.0, 1.0),
-            lambda: r.uniform(-1.0, 1.0)
-        ],
+        [f"f{c}" for c in range(columns)],
+        [lambda: r.uniform(-1.0, 1.0) for _ in range(columns)],
```
Comment (densmirn) [inline sdc/tests/tests_perf/test_perf_read_csv.py]: Why not just call `r.uniform(-1.0, 1.0)` here without lambda?
`[r.uniform(-1.0, 1.0) for _ in columns]`

Comment (PokhodenkoSA) [reply sdc/tests/tests_perf/test_perf_read_csv.py]: Because it is generators, not values.

**Comment by PokhodenkoSA** [comment]
This PR looks very big. I will split it to many smaller ones with dedicated improvements. Now I use it for prepare benchmarking report only.

**Pull request merged**